subxt/src/tx/tx_client.rs

```diff
@@ -465,4 +478,67 @@ where
         let dry_run_bytes = self.client.rpc().dry_run(self.encoded(), at).await?;
         dry_run_bytes.into_dry_run_result(&self.client.metadata())
     }
+
+    /// returns an estimate for the partial fee. There are two ways of obtaining this, method I is used here in practice. Both methods should give the exact same result though.
+    /// ## Method I: TransactionPaymentApi_query_info
+    ///
+    #[cfg_attr(doctest, doc = " ````no_test")]
+    /// ```
+    /// let encoded_with_len = self.encoded_with_len();
+    /// let RuntimeDispatchInfo{ partial_fee, ..} = self.client
+    ///  .rpc()
+    ///  .state_call_decoded::<RuntimeDispatchInfo>(
+    ///      "TransactionPaymentApi_query_info",
+    ///      Some(&encoded_with_len),
+    ///      None,
+    /// ).await?;
+    /// ```
+    ///
+    /// Here the `partial_fee` is already the result we are looking for.
+    ///
+    /// ## Method II: TransactionPaymentApi_query_fee_details
+    ///
+    /// Make a state call to "TransactionPaymentApi_query_fee_details":
+    #[cfg_attr(doctest, doc = " ````no_test")]
+    /// ```
+    /// let encoded_with_len = self.encoded_with_len();
+    /// let InclusionFee {
+    ///     base_fee,
+    ///     len_fee,
+    ///     adjusted_weight_fee,
+    /// } = self.client.rpc().state_call_decoded::<FeeDetails>(
+    ///     "TransactionPaymentApi_query_fee_details",
+    ///     Some(&encoded_with_len),
+    ///     None,
+    /// ).await?.inclusion_fee.unwrap();
+    /// let partial_fee_2 = base_fee + len_fee + adjusted_weight_fee;
+    /// ```
+    ///
+    /// Referring to this [this stackexchange answer by jsdw](https://substrate.stackexchange.com/questions/2637/determining-the-final-fee-from-a-client/4224#4224) the formula for the _partial_fee_ is:
+    ///
+    /// ```txt
+    /// partial_fee = base_fee + len_fee + (adjusted_weight_fee/estimated_weight*actual_weight)
+    /// ```
+    /// We assume here, that _estimated_weight == actual_weight_.
+    ///
+    /// So this should hold and give the same result as Method I:
+    /// ```txt
+    /// partial_fee = base_fee + len_fee + adjusted_weight_fee
+    /// ```
+    pub async fn partial_fee_estimate(&self) -> Result<u128, Error> {
```

Comment:
I think I'd get rid of most of the comment above, which is more talking about some alternates and describing how the method works, and just leave something that focuses on what the output is eg:

```
/// This returns an estimate for what the extrinsic is expected to cost to execute, less any tips.
/// The actual amount paid can vary from block to block based on node traffic and other factors.
```

(I didn't think too carefully about this wording but yeah; we can be more concise :))